tests_integration/test_nativeapp.py

```diff
@@ -564,3 +564,172 @@ def test_nativeapp_init_deploy(
                 env=TEST_ENV,
             )
             assert result.exit_code == 0
+
+
+# Tests a simple flow of executing "snow app deploy [files]", verifying that only the specified files are synced to the stage
+@pytest.mark.integration
+def test_nativeapp_init_deploy_files(
```

Comment:
The no-arg pruning behaviour for deploy isn't coverage, and needs to be. Let's make sure tests cover the full contract:

- snow app deploy (will prune)
- snow app deploy --prune (will also obviously prune)
- snow app deploy --no-prune (will obviously not prune remote)
- snow app deploy local_file (will not prune)
- snow app deploy local_file --prune/--no-prune

Reply:
(NB: the team's SOP is to not resolve comments unless they're trivially addressed so that the conversation can happen, please don't resolve immediately in the future)

IMHO these tests are important. Keep in mind that SnowflakeCLI is developed by multiple teams. Having full contract coverage in ITs serves both as a way to prevent regressions, but also as a form of executable documentation. We do leverage parts of the core platform in our logic, so unrelated changes could end up breaking our logic if it's left uncovered. We are generally thorough when it comes to testing new logic. Thoughts?

Reply:
Bumping this thread, @sfc-gh-gbloom it would be great to have the above test cases all covered, especially because we now have an external team contributing code to our NA plugin and since we do not have the same level of communication with them, these tests will help catch issues early for future development.

Reply:
Sure, I added the remaining tests.

I'm all in favor of having full test coverage for our interface, but I think that having an integration test for each permutation is excessive, especially when currently there is a lot of setup/teardown boilerplate.
We could invest in some infra to make these tests slimmer, and I think it's ok for some of the flows to be covered in unit tests.

BTW not that it's not a good goal to have, but I believe we don't have full interface coverage at the moment.